### PR TITLE
App from tmpl on subscription callback

### PR DESCRIPTION
**Pull request by alextargov**
**Description**

Changes proposed in this pull request:
- Create Application from ApplicationTemplate when a subscription callback is executed.
- When a payload is present, search for AppTemplates and Runtimes for the given criteria.

  - When an AppTemplate is found, create an App from it.
  - When a Runtime is found, subscribe the tenant to the Runtime.
  - When a Runtime and AppTemplate is found - error
  - When no Runtime nor AppTemplate is found - error

Additionally, for the initial implementation Filters for AppTemplates were introduced. After evaluation of the initial implementation was done, another approach was taken and Filters for ApplicationTemplates query were not needed but decided to leave them in the PR for future usage.

**Related issue(s)**

- https://github.tools.sap/cmp/management-plane-config/pull/421

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

File: components/director/hack/jwt_generator.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ function get_token(){
...
+    local PAYLOAD=$(echo "{ \"scopes\": \"tenant_subscription:write tenant:write fetch-request.auth:read webhooks.auth:read application.auths:read application.webhooks:read application_template:write application_template:read application_template.webhooks:read document.fetch_request:read event_spec.fetch_request:read api_spec.fetch_request:read runtime.auths:read integration_system.auths:read bundle.instance_auths:read bundle.instance_auths:read application:read automatic_scenario_assignment:write automatic_scenario_assignment:read health_checks:read application:write runtime:write label_definition:write label_definition:read runtime:read tenant:read formation:write internal_visibility:read tenant_subscription:write\", \"tenant\":\"{\\\"consumerTenant\\\":\\\"$INTERNAL_TENANT_ID\\\",\\\"externalTenant\\\":\\\"3e64ebae-38b5-46a0-b1ed-9ccee153a0ae\\\"}\" }" | base64 | tr '/+' '_-' | tr -d '=')
```
Comment (StefanCholakov) [inline components/director/hack/jwt_generator.sh:13]: **tenant_subscription:write** scope is duplicated here

File: components/director/internal/domain/apptemplate/repository_test.go [hunk 2]
```diff
@@ -308,6 +391,8 @@ func TestRepository_List(t *testing.T) {
 			*fixEntityApplicationTemplate(t, "id3", "name3"),
 		}
 
+		labelFilters := []*labelfilter.LabelFilter{labelfilter.NewForKeyWithQuery(RegionKey, "еу-1")}
```
Comment (dzahariev) [inline components/director/internal/domain/apptemplate/repository_test.go:394]: eu-1 probably, but it is Ok to be any random value

**Review by dzahariev — APPROVED** [review]

**Review by StefanCholakov — APPROVED** [review]

File: components/external-services-mock/internal/subscription/handler.go [hunk 3]
```diff
@@ -186,6 +186,11 @@ func (h *handler) createTenantRequest(httpMethod, tenantFetcherUrl, token, provi
 		}
 	}
 
+	body, err = sjson.Set(body, h.providerConfig.SubscriptionAppNameProperty, AppName)
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("An error occured when setting json value: %v", err))
+	}
+
```
[the remark this answers is not in the document]
Comment (dzahariev) [reply components/external-services-mock/internal/subscription/handler.go:193]: I think there is such negative test here: https://github.com/kyma-incubator/compass/pull/2404/files#diff-c525e781d28addb19a909410899c388b9e15d68e55c573fae407625f89274c2fR205

File: components/external-services-mock/internal/subscription/config.go [hunk 4]
```diff
@@ -34,6 +35,8 @@ type ProviderConfig struct {
 	SubdomainProperty              string `envconfig:"APP_TENANT_PROVIDER_SUBDOMAIN_PROPERTY"`
 	SubscriptionProviderIDProperty string `envconfig:"APP_TENANT_PROVIDER_SUBSCRIPTION_PROVIDER_ID_PROPERTY"`
 	ProviderSubaccountIDProperty   string `envconfig:"APP_TENANT_PROVIDER_PROVIDER_SUBACCOUNT_ID_PROPERTY"`
+	SubscribedSubaccountIDProperty string `envconfig:"APP_TENANT_PROVIDER_SUBSCRIBED_SUBACCOUNT_ID_PROPERTY"`
```
[the remark this answers is not in the document]
Comment (dzahariev) [reply components/external-services-mock/internal/subscription/config.go:38]: Removed

**Pull request merged**